CSS Grid Changes

brian-penguin opened the pull request:
Before: 
![Before](https://user-images.githubusercontent.com/7143499/88489097-39b20700-cf60-11ea-801a-f01ca0c9f969.png)

After: 
![After](https://user-images.githubusercontent.com/7143499/88489103-46365f80-cf60-11ea-829d-0ba51b0eee41.png)

After- Print-view 
![after-print-view](https://user-images.githubusercontent.com/7143499/88489110-4df60400-cf60-11ea-8a27-799aba05fde8.png)

Hi Edward! 
I've been working on getting some CSS Grid understanding for some other projects at work for a client. I figured this is a pretty good application of them. Also added some font that I enjoy (feel free to change it)

Changes: 
- Format the HTML (editor did this autotmagically) 
- Add a link to the source code 
- Add CSS Grid 
- Change font sizes to use relative sizing with the base being the Category level set to size 14px

brian-penguin committed:
Add fira mono! move some text around and link to source code

brian-penguin committed:
Add in CSS Grid!

I've been working on some stuff recently to better understand css grid
and this seems like a decent application for it.

Changes:
- Make the whole page a 3 column grid
- the first column will contain the code mirror source at 33% page width
- Second will have the formatted output and totals calculated on the
  right side
- Third is a free space that can be flexed into when the input from the
  code mirror gets large
- Update the Print view to also include font changes and avoid the
  source from the code mirror updates

brian-penguin committed:
Add a top border over the Total row

edward3h commented on index.html:
```diff
@@ -8,50 +8,51 @@
         </textarea>
         <div id="help">
           <a href="help.html" target="_blank">Open help page</a>
+          <a href="https://github.com/edward3h/simple_army_editor" target="_blank"> Source Code</a>
```
I'd prefer something like "View source in Github"

edward3h reviewed (commented):
I have a challenge: do you think you can make it so the user can select a font from a drop-down box? (I haven't yet thought about how that might work :-) I'm not entirely sure about defaulting to a monospace font.

Otherwise looks good, thanks!

brian-penguin committed:
Add a font selector!

This changes the font on the body element of the page on select

Caveat: Since this is a one way binding, refreshing the page or leaving
and revisiting will reset the font choices

brian-penguin committed:
Use localStorage to persist font family changes

why: This allows you to always use a font family of your choice and keep
it between page visits and refreshes. This family style will also work
through the print page

brian-penguin commented:
Update to allow font-family selections and persist between reloads / page visits using localStore

![take2](https://user-images.githubusercontent.com/7143499/88701564-27f36f80-d0d8-11ea-9545-5b93732682e0.png)
![take3](https://user-images.githubusercontent.com/7143499/88701574-2a55c980-d0d8-11ea-9055-573ea96156a6.png)

edward3h reviewed (approved):
Thanks for adding the font selector
